frontend/libs/studio-components/src/components/StudioPageHeader/StudioPageHeaderButton/StudioPageHeaderButton.test.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import { StudioPageHeaderButton, type StudioPageHeaderButtonProps } from './StudioPageHeaderButton';
+
+describe('StudioPageHeaderButton', () => {
+  it('should apply the correct class based on variant and color - regular dark', () => {
+    renderStudioPageHeaderButton({ variant: 'regular', color: 'dark' });
+
+    const button = screen.getByRole('button');
+    expect(button).toHaveClass('regularDark');
+  });
+
+  it('should apply the correct class based on variant and color - regular light', () => {
+    renderStudioPageHeaderButton({ variant: 'regular', color: 'light' });
+
+    const button = screen.getByRole('button');
+    expect(button).toHaveClass('regularLight');
+  });
+
+  it('should apply the correct class based on variant and color - preview dark', () => {
+    renderStudioPageHeaderButton({ variant: 'preview', color: 'dark' });
+
+    const button = screen.getByRole('button');
+    expect(button).toHaveClass('previewDark');
+  });
+
+  it('should apply the correct class based on variant and color - preview light', () => {
+    renderStudioPageHeaderButton({ variant: 'preview', color: 'light' });
+
+    const button = screen.getByRole('button');
+    expect(button).toHaveClass('previewLight');
+  });
+
```

Comment:
We should have another testing library such as Percy to test this. We are just testing that the className is current, but when the implementation of the class name changes, the tests will not fail. But, leave it for now. 😊 

But, should we add a test for the `onClick` callback?

Reply:
Great idea! Added now 😄